src/github/command.rs

```diff
@@ -0,0 +1,445 @@
+            Postponed => "postponed",
+            Closed => "closed",
+            DispositionMerge => "disposition-merge",
+            DispositionClose => "disposition-close",
+            DispositionPostpone => "disposition-postpone",
+        }
+    }
+}
+
+impl fmt::Display for Label {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        fmt.write_str(self.as_str())
+    }
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum RfcBotCommand<'a> {
+    FcpPropose(FcpDisposition),
+    FcpCancel,
+    Reviewed,
+    NewConcern(&'a str),
+    ResolveConcern(&'a str),
+    FeedbackRequest(&'a str),
+    AskQuestion {
+        teams: BTreeSet<&'a str>,
+        question: &'a str,
+    },
+}
+
+impl<'a> RfcBotCommand<'a> {
+    pub fn from_str_all(command: &'a str) -> impl Iterator<Item = RfcBotCommand<'a>> {
+        // Get the tokens for each command line (starts with a bot mention)
+        command.lines()
+               .filter(|&l| l.starts_with(RFC_BOT_MENTION))
+               .map(from_invocation_line)
+               .filter_map(Result::ok)
+    }
+
+}
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum FcpDisposition {
+    Merge,
+    Close,
+    Postpone,
+}
+
+const FCP_REPR_MERGE: &'static str = "merge";
+const FCP_REPR_CLOSE: &'static str = "close";
+const FCP_REPR_POSTPONE: &'static str = "postpone";
+
+impl FcpDisposition {
+    pub fn repr(self) -> &'static str {
+        match self {
+            FcpDisposition::Merge => FCP_REPR_MERGE,
+            FcpDisposition::Close => FCP_REPR_CLOSE,
+            FcpDisposition::Postpone => FCP_REPR_POSTPONE,
+        }
+    }
+
+    pub fn from_str(string: &str) -> DashResult<Self> {
+        Ok(match string {
+            FCP_REPR_MERGE => FcpDisposition::Merge,
+            FCP_REPR_CLOSE => FcpDisposition::Close,
+            FCP_REPR_POSTPONE => FcpDisposition::Postpone,
+            _ => throw!(DashError::Misc(None)),
+        })
+    }
+
+    pub fn label(self) -> Label {
+        match self {
+            FcpDisposition::Merge => Label::DispositionMerge,
+            FcpDisposition::Close => Label::DispositionClose,
+            FcpDisposition::Postpone => Label::DispositionPostpone,
+        }
+    }
+}
+
+/// Parses the text of a subcommand.
+fn parse_command_text<'a>(command: &'a str, subcommand: &'a str) -> &'a str {
+    let name_start = command.find(subcommand).unwrap() + subcommand.len();
+    command[name_start..].trim()
+}
+
+fn strip_prefix<'h>(haystack: &'h str, prefix: &str) -> &'h str {
+    haystack.find(prefix)
+            .map(|idx| &haystack[idx + prefix.len()..])
+            .unwrap_or(haystack)
+            .trim()
+}
+
+fn match_team_candidate(team_candidate: &str) -> Option<&'static TeamLabel> {
+    #[cfg(not(test))]
+    let setup = &*teams::SETUP;
+    #[cfg(test)]
+    let setup = &*teams::test::TEST_SETUP;
+
+    setup.teams().find(|&(label, team)| {
+        strip_prefix(&label.0, "T-") == strip_prefix(team_candidate, "T-") ||
```

Comment:
Can we support both the `T-name` and `name` forms here?

Reply:
That's what the line does :)